amoro-common/src/main/java/org/apache/amoro/resource/ExternalResourceContainer.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.amoro.resource;
+
+import org.apache.amoro.process.AmoroProcess;
+import org.apache.amoro.process.TableProcessState;
+
+/**
+ * ExternalResourceContainer is the key interface for the AMS framework to interact with the
+ * external resource. Typically, it is used to submit and release resource on YARN/K8S.
+ */
+public interface ExternalResourceContainer extends ResourceContainer {
```

Comment:
Do you mean the life cycle of `ExternalResourceContainer` is not managed by AMS?

> Typically, it is used to submit and release resource on YARN/K8S.

I don't quite understand here. If AMS itself runs on K8S, then `InternalResourceContainer` will also run on K8S?What is the difference between `InternalResourceContainer` and `ExternalResourceContainer`?

Reply:
For InternalContainer executors，they recieve tasks scheduled by AMS, and released by AMS when necessary. For ExternalContainer executors, they recieved tasks by some JobManager or application master. The true diff is that ResourceManager(response for task scheduling and managing lifetime of executors) is inside AMS or outside

Reply:
PTAL